client/src/app/views/admin/views/admin-profile/admin-profile-settings/admin-profile-settings.component.ts

```diff
@@ -31,31 +32,49 @@ export class AdminProfileSettingsComponent implements OnInit {
     public configService: ConfigService
   ) { }
 
-  async ngOnInit () {
+  async ngOnInit() {
     this.profileId = this.profileService.profileId;
     this.profiles = await this.dataService.getProfiles();
+    this.updateProfilesValidAsParent();
+    this.pboCategories = await this.adminService.getPboCategories();
+
     this.profileId.subscribe(profileId => {
-      if(profileId) this.loadProfile(profileId)
+      if (profileId) this.loadProfile(profileId)
     });
   }
 
   async loadProfile(profileId: number) {
     this.profile = await this.adminService.getProfile(profileId);
+
+    // Use default PBO category if not set
+    if (this.profile.type === "pbo" && this.pboCategories.length > 0) {
+      this.profile.pboCategoryId ??= this.pboCategories[0].pboCategoryId;
+    }
+
+    this.updateProfilesValidAsParent();
   }
 
-  async reloadProfile(){
-    this.profile = await this.adminService.getProfile(this.profile.id);
+  async reloadProfile() {
+    this.loadProfile(this.profile.id);
     this.profileService.setProfile(this.profile);
   }
 
   async saveProfile(form: NgForm) {
     const data = form.value;
-    if(data.parent == "null") data.parent = null;
+
+    if (data.parent == "null") data.parent = null;
+
     await this.adminService.saveProfile(this.profile.id, data)
+
     this.reloadProfile();
 
     this.toastService.toast("Uloženo.", "notice")
+  }
 
+  updateProfilesValidAsParent() {
+    this.profilesValidAsParent = this.profiles
+      ? this.profiles.filter(p => p.parent == null && (this.profile == null || this.profile.id !== p.id))
```

Comment:
add coment explaining the second half of the condition

Reply:
Done